File: app/src/main/java/eu/sesma/dagger/platform/DetailActivityModule.java
package eu.sesma.dagger.platform;

import eu.sesma.dagger.di.PerActivity;
import eu.sesma.dagger.ui.detail.DetailActivity;

import android.app.Activity;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class DetailActivityModule {
    @Provides
    @PerActivity
    static Activity provideActivity(DetailActivity activity) {
        return activity;
    }
}

File: module_detail/src/main/java/eu/sesma/dagger/detail/di/DetailActivityModule.kt
package eu.sesma.dagger.detail.di

import eu.sesma.dagger.core.di.ActivityProvider
import eu.sesma.dagger.detail.DetailActivity
import eu.sesma.dagger.detail.DetailNavigator
import eu.sesma.dagger.detail.DetailPresenter
import org.koin.android.ext.koin.androidContext
import org.koin.core.qualifier.named
import org.koin.dsl.module


val detailActivityModule = module {
    scope(named<DetailActivity>()) {
        scoped {
            DetailNavigator(
                    context = androidContext(),
                    coreCollaborator = get(),
                    activity = (androidContext() as ActivityProvider).activeActivity
            )
        }

        scoped {
            DetailPresenter(
                    navigator = get(),
                    coreSingletonCollaborator = get(),
                    coreCollaborator = get()
            )
        }
    }
}